site/public/js/homepage.js

```diff
@@ -0,0 +1,58 @@
+
+/**
+ * Get the UTC offset of the user's local time zone
+ *
+ * @return {string} of the user's local time zone UTC offset, for example for example '+9:30' or '-4:00'
+ */
+function getCurrentUTCOffset() {
+    const date = new Date();
+    const sign = (date.getTimezoneOffset() > 0) ? '-' : '+';
+    const offset = Math.abs(date.getTimezoneOffset());
+    let hours = Math.floor(offset / 60);
+    hours = (hours < 10 ? `0${hours}` : hours);
+    return `${sign + hours}:00`;
+}
+
+// once the homepage loads, does a timezone check against the user's local timezone and
+//if the user set a timezone,  warning will pop up if there is a timezone mismatch
+window.addEventListener('load', function () {
+    console.log("home page loaded");
+
+    $.getJSON({
```

Comment:
See prior comment about AJAX request being unnecessary.  This code could just check the value of a JS constant set via Twig.